File: assignments/myanmar_digit_calculator/javascript/myanmar_digit_calculator_test_runner.js
const assert = require('assert');
const { MyanmarDigitCalculator } = require('./MyanmarCalculator.js');

function runTests() {
    const solver = new MyanmarDigitCalculator();
    const testCases = [
        { n1: "၁၂၃", n2: "၄၅၆", op: '+', expected: "၅၇၉" },
        { n1: "၁၀၀", n2: "၁၀", op: '-', expected: "၉၀" },
        { n1: "၂၀", n2: "၅", op: '*', expected: "၁၀၀" },
        { n1: "၂၁", n2: "၅", op: '/', expected: "၄" },
        { n1: "၁၀", n2: "၀", op: '/', expected: "Error: Division by zero" },
        { n1: "၀", n2: "၅", op: '+', expected: "၅" },
        { n1: "၅", n2: "၀", op: '*', expected: "၀" },
        { n1: "၉၈၇", n2: "၁၂၃", op: '+', expected: "၁၁၁၀" },
        { n1: "၁၀", n2: "၂၀", op: '-', expected: "-၁၀" }, // Test negative result
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for MyanmarCalculator (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "";
        try {
            actual = solver.calculate(tc.n1, tc.n2, tc.op);
            assert.strictEqual(actual, tc.expected);
            report.push(`Test Case #${i + 1}: Input '${tc.n1}' ${tc.op} '${tc.n2}' -> ✅ PASS`);
            passedCount++;
        } catch (error) {
             report.push(`Test Case #${i + 1}: Input '${tc.n1}' ${tc.op} '${tc.n2}' -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${actual}"`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();

File: assignments/myanmar_digit_calculator/python/myanmar_digit_calculator_test_runner.py
import sys
from myanmar_digit_calculator import MyanmarCalculator

def run_tests():
    solver = MyanmarCalculator()
    test_cases = [
        ("၁၂၃", "၄၅၆", '+', "၅၇၉"),
        ("၁၀၀", "၁၀", '-', "၉၀"),
        ("၂၀", "၅", '*', "၁၀၀"),
        ("၂၁", "၅", '/', "၄"),
        ("၁၀", "၀", '/', "Error: Division by zero"),
        ("၀", "၅", '+', "၅"),
        ("၅", "၀", '*', "၀"),
        ("၉၈၇", "၁၂၃", '+', "၁၁၁၀"),
        ("၁၀", "၂၀", '-', "-၁၀") # Python's str() handles negative correctly
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for MyanmarCalculator (Python) =====")

    for i, (n1, n2, op, expected) in enumerate(test_cases):
        actual = ""
        try:
            actual = solver.calculate(n1, n2, op)
            report.append(f"Test Case #{i + 1}: Input '{n1}' {op} '{n2}'")
            report.append(f" -> Expected: \"{expected}\", Got: \"{actual}\"")

            if actual == expected:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(f"Test Case #{i + 1}: Input '{n1}' {op} '{n2}'")
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()